### PR TITLE
XCTest: update shebang line to name python3 instead of python generically.

### PR DESCRIPTION
**Pull request by fischman-bcny**
Motivated by [failure](https://ci.swift.org/job/swift-corelibs-foundation-PR-macOS/221/execution/node/43/log/) from https://github.com/apple/swift-corelibs-foundation/pull/4740's test runs, which ended with:
```
+ /Users/ec2-user/jenkins/workspace/swift-corelibs-foundation-PR-macOS/branch-main/swift-corelibs-xctest/build_script.py --swiftc=/Users/ec2-user/jenkins/workspace/swift-corelibs-foundation-PR-macOS/branch-main/build/Ninja-ReleaseAssert/swift-macosx-x86_64/bin/swiftc --build-dir=/Users/ec2-user/jenkins/workspace/swift-corelibs-foundation-PR-macOS/branch-main/build/Ninja-ReleaseAssert/xctest-macosx-x86_64 --foundation-build-dir=/Users/ec2-user/jenkins/workspace/swift-corelibs-foundation-PR-macOS/branch-main/build/Ninja-ReleaseAssert/foundation-macosx-x86_64 --swift-build-dir=/Users/ec2-user/jenkins/workspace/swift-corelibs-foundation-PR-macOS/branch-main/build/Ninja-ReleaseAssert/swift-macosx-x86_64 --release
env: python: No such file or directory
```

### CODE REVIEW

**Comment by compnerd** [comment]
CC: @edymtt

**Comment by compnerd** [comment]
CC: @dduan

**Review by dduan — APPROVED** [review]

**Comment by compnerd** [comment]
@swift-ci please test macOS platform

**Comment by compnerd** [comment]
@swift-ci please test Windows platform

**Review by edymtt — APPROVED** [review]

**Comment by fischman-bcny** [comment]
@dduan @edymtt macOS CI failure seems unrelated to this PR. Is it possible to merge this to unblock https://github.com/apple/swift-corelibs-foundation/pull/4740 ?

(I've tried & failed to repro the test failures on darwin; on my machine I see _different_ unrelated failures with the xctest test subprocesses exiting -5 whereas CI has them exiting -4...)

**Comment by shahmishal** [comment]
We need to update the python version here. 
https://github.com/apple/swift-corelibs-xctest/blob/main/Tests/Functional/xctest_checker/xctest_checker.py

**Comment by fischman-bcny** [comment]
@shahmishal I had done that in the first version of this PR (https://github.com/apple/swift-corelibs-xctest/commit/2009414b2202ed8bbe0bc5cc4345dc4d6c3b05b4) but that also failed CI. 
Are you suggesting xctest_checker.py should have python3 but xctest_checker/main.py should _not_?

(happy to update this PR to be any of these)

**Comment by shahmishal** [comment]
I think all of the files should use python3, as we no longer have python2 on the CI nodes.

**Comment by fischman-bcny** [comment]
@shahmishal ok, done.

**Comment by shahmishal** [comment]
@swift-ci test macOS

**Comment by fischman-bcny** [comment]
@shahmishal We're back to the CI failing on unrelated issues. Since we know that the CI machines don't have a `python` binary and do have a `python3` binary, and since this is holding up submissions to the foundation repo, can we merge this PR now?

**Comment by fischman-bcny** [comment]
Same command that failed on CI succeeded on my machine: https://gist.githubusercontent.com/fischman-bcny/008a14e4a2afe42a2767dd97ec3cc636/raw/05c5125ba49133bc64e5bc4b0a029954f8f65e81/gistfile0.txt

@shahmishal is there a way I can ssh into the CI machine to investigate its failures? Or any other suggestions for unblocking this PR?

**Comment by fischman-bcny** [comment]
Closing this PR since https://github.com/apple/swift/pull/65958 merged.

**Pull request closed without merging**
